sound/soc/sof/intel/pci-tgl.c

```diff
@@ -298,7 +298,7 @@ static const struct pci_device_id sof_pci_ids[] = {
 	{ PCI_DEVICE_DATA(INTEL, HDA_ADL_PX, &adl_desc) },
 	{ PCI_DEVICE_DATA(INTEL, HDA_RPL_M, &rpl_desc) },
 	{ PCI_DEVICE_DATA(INTEL, HDA_RPL_PX, &rpl_desc) },
-	{ PCI_DEVICE_DATA(INTEL, HDA_ADL_N, &adl_n_desc) },
+	{ PCI_DEVICE_DATA(INTEL, HDA_ADL_N, &adln_desc) },
```

Comment:
We seem to have two conventions on naming, not sure why we removed the underscore in the descriptor and added it in the firmware/path name...

````c
	[SOF_IPC_TYPE_4] = "intel/sof-ipc4-lib/tgl-h",
	[SOF_IPC_TYPE_3] = "sof-tgl-h.ri",
````

Reply:
this is the struct name in kernel, don't ask me why it is like this, but it is since like forever, before even ADL-N, aka, adln, aka adl-n
The struct names are w/o underscore for sure in this file.

Reply:
yeah, I guess the reason is lost to history.